internal/pkg/aws/cloudwatch/cloudwatch.go

```diff
@@ -101,6 +101,31 @@ func (cw *CloudWatch) AlarmStatus(alarms []string) ([]AlarmStatus, error) {
 	return alarmStatus, nil
 }
 
+// AlarmStatusesFromNamePrefix returns the status of all alarms whose names begin with the given prefix.
+func (cw *CloudWatch) AlarmStatusesFromNamePrefix(prefix string) ([]AlarmStatus, error) {
+	alarmResp := &cloudwatch.DescribeAlarmsOutput{}
+	var err error
+	var alarmStatuses []AlarmStatus
+	for {
+		alarmResp, err = cw.client.DescribeAlarms(&cloudwatch.DescribeAlarmsInput{
+			AlarmNamePrefix: aws.String(prefix),
+			NextToken:       alarmResp.NextToken,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("describe CloudWatch alarms: %w", err)
+		}
+		if alarmResp == nil {
+			return nil, nil
+		}
```

Comment:
Should we `break` instead?

Reply:
Is there any benefit to using `break` instead? I figured `return` exits the entire method earlier....

Reply:
Break returns `alarmStatuses` and `nil`, which is the same as returning `alarmStatuses` and `nil` here. One scenario is `alarmResp` was not `nil` before and `alarmStatuses` got some values already, and in one page loop however, we get an empty `alarmStatuses` and in your current logic, it will return `nil` and `nil`.

Reply:
I didn't think that scenario was possible, but sure, I can change it just in case!